tinkerbell/boots/templates/deployment.yaml

```diff
@@ -29,26 +33,42 @@ spec:
       containers:
         - image: {{ .Values.image }}
           imagePullPolicy: {{ .Values.imagePullPolicy }}
-          {{- if .Values.args }}
-          args: 
-          {{- range .Values.args }}
+          args:
+            - -dhcp-addr={{ printf "%v:%v" .Values.dhcp.ip .Values.dhcp.port }}
+            - -ipxe-tftp-addr={{ printf "%v:%v" .Values.tftp.ip .Values.tftp.port }}
+            - -http-addr={{ printf "%v:%v" .Values.http.ip .Values.http.port }}
+            - -syslog-addr={{ printf "%v:%v" .Values.syslog.ip .Values.syslog.port }}
+          {{- range .Values.additionalArgs }}
             - {{ . }}
           {{- end }}
-          {{- end }}
           env:
             - name: TRUSTED_PROXIES
-              value: {{ required "missing trustedProxies" .Values.trustedProxies | quote }}
-            {{- range $i, $env := .Values.env }}
-            - name: {{ $env.name | quote }}
-              value: {{ $env.value | quote }}
+              value: {{ required "missing trustedProxies" ( join "," .Values.trustedProxies | quote ) }}
+            - name: DATA_MODEL_VERSION
+              value: "kubernetes"
+            - name: FACILITY_CODE
+              value: "lab1"
+            - name: MIRROR_BASE_URL
+              value: {{ printf "%v://%v:%v" .Values.osieBase.protocol $osieBaseIp .Values.osieBase.port | quote }}
+            - name: PUBLIC_IP
+              value: {{ $remoteDhcpIp | quote }}
+            - name: PUBLIC_SYSLOG_FQDN
+              value: {{ $remoteSyslogIp | quote }}
+            - name: TINKERBELL_GRPC_AUTHORITY
+              value: {{ printf "%s:%v" $tinkServerIp .Values.tinkServer.port | quote }}
+            - name: TINKERBELL_TLS
+              value: {{ .Values.tinkServer.tls | quote }}
+            - name: BOOTS_EXTRA_KERNEL_ARGS
+              value: {{ join " " ( append .Values.additionlKernelArgs ( printf "tink_worker_image=%s" ( required "missing tinkWorkerImage" .Values.tinkWorkerImage ) ) ) | quote }}
+            - name: BOOTS_LOG_LEVEL
+              value: {{ .Values.logLevel | quote }}
+            {{- range .Values.additionalEnv }}
+            - name: {{ .name | quote }}
+              value: {{ .value | quote }}
             {{- end }}
           {{- if not .Values.hostNetwork }}
           ports:
-            {{- range .Values.ports }}
-            - containerPort: {{ .targetPort }}
-              name: {{ .name }}
-              protocol: {{ .protocol }}
-            {{- end }}
+            {{- include "boots.ports" ( merge .Values ( dict "PortKey" "containerPort" ) ) | indent 12 }}
```

Comment:
With `boots.hostNetwork: false` I was getting an error `helm install`-ing. It was complaining about 
```bash
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error validating "": error validating data: [ValidationError(Deployment.spec.template.spec.containers[0].ports[0]): unknown field "port" in io.k8s.api.core.v1.ContainerPort, ValidationError(Deployment.spec.template.spec.containers[0].ports[0]): missing required field "containerPort" in io.k8s.api.core.v1.ContainerPort, ValidationError(Deployment.spec.template.spec.containers[0].ports[1]): unknown field "port" in io.k8s.api.core.v1.ContainerPort, ValidationError(Deployment.spec.template.spec.containers[0].ports[1]): missing required field "containerPort" in io.k8s.api.core.v1.ContainerPort, ValidationError(Deployment.spec.template.spec.containers[0].ports[2]): unknown field "port" in io.k8s.api.core.v1.ContainerPort, ValidationError(Deployment.spec.template.spec.containers[0].ports[2]): missing required field "containerPort" in io.k8s.api.core.v1.ContainerPort, ValidationError(Deployment.spec.template.spec.containers[0].ports[3]): unknown field "port" in io.k8s.api.core.v1.ContainerPort, ValidationError(Deployment.spec.template.spec.containers[0].ports[3]): missing required field "containerPort" in io.k8s.api.core.v1.ContainerPort]
```
It looks like there is some weird overwrite behavior among all the `{{- include "boots.ports"` lines across charts. When i changed the order of the `merge` the `helm install` worked for me. So 
```bash
- {{- include "boots.ports" ( merge .Values ( dict "PortKey" "containerPort" ) ) | indent 12 }}
+ {{- include "boots.ports" ( merge ( dict "PortKey" "containerPort" ) .Values ) | indent 12 }}
```

Reply:
Thanks. Should be fixed.